src/module.c

```diff
@@ -12184,12 +12197,8 @@ int moduleLoad(const char *path, void **module_argv, int module_argc, int is_loa
         serverLog(LL_WARNING,
             "Module %s initialization failed. Module not loaded",path);
         if (ctx.module) {
-            moduleUnregisterCommands(ctx.module);
-            moduleUnregisterSharedAPI(ctx.module);
-            moduleUnregisterUsedAPI(ctx.module);
+            moduleUnregisterCleanup(ctx.module);
             moduleRemoveCateogires(ctx.module);
-            moduleRemoveConfigs(ctx.module);
-            moduleUnregisterAuthCBs(ctx.module);
             moduleFreeModuleStructure(ctx.module);
```

Comment:
The following is added here:
```
moduleFreeAuthenticatedClients
moduleUnsubscribeNotifications
moduleUnregisterFilters
moduleUnsubscribeAllServerEvents
```

Only moduleFreeAuthenticatedClients has no test cases added (it looks fine)